pkg/cmd/cli/describe/deployments.go

```diff
@@ -93,19 +93,30 @@ func (d *DeploymentConfigDescriber) Describe(namespace, name string, settings kc
 			if err == nil {
 				sorted := deploymentsHistory.Items
 				sort.Sort(sort.Reverse(rcutils.OverlappingControllers(sorted)))
+				counter := 0
 				for _, item := range sorted {
 					if item.Name != deploymentName && deploymentConfig.Name == deployutil.DeploymentConfigNameFor(&item) {
 						header := fmt.Sprintf("Deployment #%d", deployutil.DeploymentVersionFor(&item))
 						printDeploymentRc(&item, d.kubeClient, out, header, false)
+						counter++
+					}
+					// Only show last three deployments here
+					if counter == 2 {
+						break
 					}
 				}
 			}
 		}
 
 		// Events
 		if events, err := d.kubeClient.Events(deploymentConfig.Namespace).Search(deploymentConfig); err == nil && events != nil {
+			latestDeploymentEvents := &kapi.EventList{Items: []kapi.Event{}}
+			// Only show last 5 events here
```

Comment:
@kargakis can we make the number of events that are shown configurable (DescriberSettings#NumOfEvents), so users that are interested in "older" events can use that option (and we can default to 8-10)